expression/generator/string_vec.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build ignore
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"go/format"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"text/template"
+
+	. "github.com/pingcap/tidb/expression/generator/helper"
+)
+
+const header = `// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by go generate in expression/generator; DO NOT EDIT.
+
+package expression
+`
+
+const newLine = "\n"
+
+const builtinStringImports = `import (
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+)
+`
+
+var builtinStringVecTpl = template.Must(template.New("").Parse(`
+// vecEvalInt evals FIELD(str,str1,str2,str3,...).
+// See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_field
+func (b *builtinField{{ .TypeName }}Sig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	buf0, err := b.bufAllocator.get(types.ET{{ .ETName }}, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[0].VecEval{{ .TypeName }}(b.ctx, input, buf0); err != nil {
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ET{{ .ETName }}, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+{{ if .Fixed }}
+	arg0 := buf0.{{ .TypeNameInColumn }}s()
+{{ end }}
+	result.ResizeInt64(n, false)
+	i64s := result.Int64s()
+	for i := 0; i < n; i++ {
+		i64s[i] = 0
+	}
+	for i := 1; i < len(b.args); i++ {
+		if err := b.args[i].VecEval{{ .TypeName }}(b.ctx, input, buf1); err != nil {
+			return err
+		}
+{{ if .Fixed }}
+		arg1 := buf1.{{ .TypeNameInColumn }}s()
+{{ end }}
+		for j := 0; j < n; j++ {
+			if buf0.IsNull(j) || buf1.IsNull(j) || i64s[j] > 0 {
```

Comment:
For better logical shortcut.

```suggestion
			if  i64s[j] > 0 || buf0.IsNull(j) || buf1.IsNull(j) {
```

Reply:
ok